hw/ip/kmac/dv/kmac_base_sim_cfg.hjson

```diff
@@ -39,14 +32,27 @@
   // Default iterations for all tests - each test entry can override this.
   reseed: 50
 
+  build_modes: [
+    {
+      name: enable_masking
+      build_opts: ["-pvalue+tb.EnMasking=1"]
```

Comment:
Passing compile-time switches over the command line, such as `+define+foo=bar` are standardized across all tools. Unfortunately, its not the same for elab-time parameter overrides - each tool handles that differently. 

Please file a bug on DV flow to address this though. We could create a separate `parameter_overrides: []` variable in the HJSon that is eventually expanded into each tool's `build_opts`, such that it prepends the tool-specific switch name correctly.  
 
Xcelium will fail here because the corresponding option for it is `-defparam ent1.I1.param1=50`. 

In the meantime, this can be addressed like this:
```
  build_modes: [
    {
      name: vcs_mask_enabled_mode
      build_opts: ["-pvalue_tb.EnMasking=1"]
      ...
    }
    {
      name: xcelium_mask_enabled_mode
      build_opts: ["-defparam tb.EnMasking=1"]
      ...
    }
  ]

  // Enable this mode like this:
  en_build_modes: ["{tool}_mask_enabled_mode"]
```

Reply:
That's a good point, thanks for catching that. I'll file an issue for this.
In the meantime i've updated the build_modes accordingly for both VCS and Xcelium.

Reply:
Instead of changing parameter value, this can be achieved by passing a define, which might be easier.

Reply:
Yes, macro define is also a good idea.

Reply:
yes, agreed using a macro define here instead would make it simpler. thanks for the suggestion!